projects/ngx-device-detector/src/lib/device-detector.service.ts

```diff
@@ -153,13 +166,25 @@ export class DeviceDetectorService {
    * @returns whether the current device is a mobile
    */
   public isMobile(userAgent = this.userAgent): boolean {
+    if (this.cache.isMobile !== null && userAgent === this.userAgent) {
+      return this.cache.isMobile;
+    }
+    let parser = new DeviceDetectorParser();
     if (this.isTablet(userAgent)) {
       return false;
     }
-    const match = Object.keys(Constants.MOBILES_RE).find(mobile => {
-      return this.reTree.test(userAgent, Constants.MOBILES_RE[mobile]);
+    const match = Object.keys(parser.MOBILES_RE).find(mobile => {
```

Comment:
what about using array.some instead of array.find?
`const match = Object.keys(parser.MOBILES_RE).some(mobile => this.reTree.test(userAgent, parser.MOBILES_RE[mobile]));`